pkg/api/errors/pubsub.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import (
+	"fmt"
+	"net/http"
+
+	grpcCodes "google.golang.org/grpc/codes"
+
+	kiterrors "github.com/dapr/kit/errors"
+)
+
+const (
+	PostFixNameEmpty = "NAME_EMPTY"
+)
+
+func PubSubNotFound(name string, pubsubType string, metadata map[string]string) error {
```

Comment:
Could we remove `PubSub` prefix as it is in pubsub.go?

Reply:
We could make the errors more generic, but then we would have to have more parameters in the function since different APIs have different legacy tags or error codes in general that users depend on. We can't change those legacy values because we don't want to break users. For example, [StateStoreNotFound](https://github.com/dapr/dapr/pull/7257/files#diff-8d5b272f8b9930484ae10ac077dc7ee4c0656649270f3b1f7b3054b80ea92130R37), and PubSubNotFound return different values to users for legacy tags, http codes, etc. 

@elena-kolevska what are your thoughts? We would have to coordinate this change and try to combine as many as we could by making the functions general and pass in the values we need.

Reply:
Yes, exactly, the NotFound errors for different components are different because we include the component name and details.
I think merging them into a single, parametrised function will complicate the developer experience, without a big benefit.

Reply:
> Could we remove PubSub prefix as it is in pubsub.go?

Its in `pubsub.go` but the different files containing the error codes are under the same package, so method names would conflict.

Reply:
I went ahead and made a generic `NotFound` && `NotConfigured` func to be used by both state store and pubsub

Reply:
generic errors can be found in: `pkg/api/errors/errors.go`